sunpy/map/mapbase.py

```diff
@@ -1741,36 +1746,36 @@ def draw_rectangle(self, bottom_left, width: u.deg, height: u.deg, axes=None, **
 
         Notes
         -----
-
         Extra keyword arguments to this function are passed through to the
         `~matplotlib.patches.Rectangle` instance.
-
         """
+        if width is None or height is None:
+            bottom_left, top_right = get_rectangle_coordinates(bottom_left,
+                                                               top_right=top_right,
+                                                               width=width,
+                                                               height=height)
+
+            width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
+            height = Latitude(top_right.spherical.lat - bottom_left.spherical.lat)
```

Comment:
```suggestion
        bottom_left, top_right = get_rectangle_coordinates(bottom_left,
                                                           top_right=top_right,
                                                           width=width,
                                                           height=height)

        width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
        height = Latitude(top_right.spherical.lat - bottom_left.spherical.lat)
```
Now that `get_rectangle_coordinates()` provides validation on the inputs, we should always call it in all situations (even though it may result in unnecessary computations).